autoreject/autoreject.py

```
@@ -599,8 +670,13 @@ def _get_epochs_interpolation(self, epochs, labels,
                     interp_chs_mask = labels[epoch_idx] == 1
                 else:
                     # get peak-to-peak for channels in that epoch
-                    data = epochs[epoch_idx].get_data(**_GDKW)[0]
-                    peaks = np.ptp(data, axis=-1)
+                    if data_local is not None:
+                        epoch_data = data_local[epoch_idx]
+                    else:
+                        # Fallback: should not happen if needs_ptp logic is correct
```

raise error if it should not happen? Your fix may silently suppress another issue ...

---

Good catch, thank you! Changed to raise an explicit error:

```python
else:
    # This should not happen if needs_ptp logic is correct
    raise RuntimeError(
        "Unexpected fallback to individual epoch data extraction. "
        "This indicates a bug in needs_ptp computation logic."
    )
```